src/m365/planner/commands/task/task-add.ts

```diff
@@ -0,0 +1,390 @@
+  title: string;
+  startDateTime?: string;
+  dueDateTime?: string;
+  percentComplete?: number;
+  assignedToUserIds?: string;
+  assignedToUserNames?: string;
+  description?: string;
+  orderHint?: string;
+}
+
+class PlannerTaskAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.TASK_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a new Microsoft Planner Task';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.planId = typeof args.options.planId !== 'undefined';
+    telemetryProps.planName = typeof args.options.planName !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    telemetryProps.bucketId = typeof args.options.bucketId !== 'undefined';
+    telemetryProps.bucketName = typeof args.options.bucketName !== 'undefined';
+    telemetryProps.startDateTime = typeof args.options.startDateTime !== 'undefined';
+    telemetryProps.dueDateTime = typeof args.options.dueDateTime !== 'undefined';
+    telemetryProps.percentComplete = typeof args.options.percentComplete !== 'undefined';
+    telemetryProps.assignedToUserIds = typeof args.options.assignedToUserIds !== 'undefined';
+    telemetryProps.assignedToUserNames = typeof args.options.assignedToUserNames !== 'undefined';
+    telemetryProps.description = typeof args.options.description !== 'undefined';
+    telemetryProps.orderHint = typeof args.options.orderHint !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this.getPlanId(args)
+      .then(planId => {
+        args.options.planId = planId;
+        return this.getBucketId(args, planId);
+      })
+      .then(bucketId => {
+        args.options.bucketId = bucketId;
+        return this.generateUserAssignments(args);
+      })
+      .then(assignments => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/tasks`,
+          headers: {
+            'accept': 'application/json;odata.metadata=none'
+          },
+          responseType: 'json',
+          data: {
+            planId: args.options.planId,
+            bucketId: args.options.bucketId,
+            title: args.options.title,
+            startDateTime: args.options.startDateTime,
+            dueDateTime: args.options.dueDateTime,
+            percentComplete: args.options.percentComplete,
+            assignments: assignments,
+            orderHint: args.options.orderHint
+          }
+        };
+
+        return request.post(requestOptions) as PlannerTask;
+      })
+      .then(newTask => { 
+        const taskId = newTask.id as string;
+  
+        if (args.options.description) {
+          return this.getTaskDetailsEtag(taskId)
```

Comment:
Let's not nest promises because it makes it harder to follow the code. In this case, it would be easier to extract this logic to a separate function that returns an early `Promise.resolve()` if description was not set, and otherwise, updates the task with the specified description.